File: src/Entity/Promotion.php
<?php

namespace App\Entity;

use App\Repository\PromotionRepository;
use Doctrine\ORM\Mapping as ORM;
use App\Entity\Artwork;


#[ORM\Entity(repositoryClass: PromotionRepository::class)]
class Promotion
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column]
    private ?int $remise = null;

    #[ORM\Column(length: 255)]
    private ?string $type = null;

    #[ORM\ManyToOne(inversedBy: 'promotions')]
    #[ORM\JoinColumn(name: 'id_artwork', referencedColumnName: 'id_artwork')]
    private ?Artwork $id_artwork = null;

    #[ORM\Column(length: 255)]
    private ?string $nom_artwork = null;

    #[ORM\Column(length: 255)]
    private ?string $prix_artwork = null;

    public function getId(): ?int
    {
        return $this->id;
    }
   

    public function getRemise(): ?int
    {
        return $this->remise;
    }

    public function setRemise(int $remise): self
    {
        $this->remise = $remise;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getidArtwork(): ?Artwork
    {
        return $this->id_artwork;
    }
    
    public function getnomArtwork(): ?string
    {
        return $this->nom_artwork;
    }
  

    public function setArtwork(?Artwork $artwork): self
    {
        $this->id_artwork = $artwork;

        return $this;
    }

    public function setNomArtwork(string $nomArtwork): self
    {
        $this->nom_artwork = $nomArtwork;

        return $this;
    }

    public function getPrix(): ?string
    {
        return $this->prix_artwork;
    }

    public function setPrix(string $prix): self
    {
        $this->prix_artwork = $prix;

        return $this;
    }
}

File: Hamza/src/tn/gestion/artfulio/entites/Promotion.java
    package tn.gestion.artfulio.entites;

public class Promotion {
    private int id;
    private int remise ;
    private String type;
    private String nom_artwork;
    private String prix_artwork;
    private int id_artwork;

    public Promotion(int id, int remise, String type, String nom_artwork, String prix_artwork, int id_artwork) {
        this.id = id;
        this.remise = remise;
        this.type = type;
        this.nom_artwork = nom_artwork;
        this.prix_artwork = prix_artwork;
        this.id_artwork = id_artwork;
    }

    public Promotion() {
    }

    public Promotion(int remise, String type, String nom_artwork, String prix_artwork, int id_artwork) {
        this.remise = remise;
        this.type = type;
        this.nom_artwork = nom_artwork;
        this.prix_artwork = prix_artwork;
        this.id_artwork = id_artwork;
    }

    public Promotion(int id, int remise, String type, String nom_artwork, String prix_artwork) {
        this.id = id;
        this.remise = remise;
        this.type = type;
        this.nom_artwork = nom_artwork;
        this.prix_artwork = prix_artwork;
    }
    
    

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getRemise() {
        return remise;
    }

    public void setRemise(int remise) {
        this.remise = remise;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getNom_artwork() {
        return nom_artwork;
    }

    public void setNom_artwork(String nom_artwork) {
        this.nom_artwork = nom_artwork;
    }

    public String getPrix_artwork() {
        return prix_artwork;
    }

    public void setPrix_artwork(String prix_artwork) {
        this.prix_artwork = prix_artwork;
    }

    public int getId_artwork() {
        return id_artwork;
    }

    public void setId_artwork(int id_artwork) {
        this.id_artwork = id_artwork;
    }

    
}
